infra/modules/azure_app_service_plan_autoscaler/variables.tf

```diff
@@ -0,0 +1,122 @@
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "app_service_name" {
+  type        = string
+  description = "Set name of the App Service to monitor"
+  default     = null
+}
+
+variable "function_app_name" {
+  type        = string
+  description = "Set the name of the Function App to monitor"
+  default     = null
+}
+
```

Comment:
what about adding some condition (exactly one between function_app_name OR app_service_name must be set):

```hcl
variable "app_service_name" {
  description = "The name of the App Service"
  type        = string
  default     = null
}

variable "function_app_name" {
  description = "The name of the Function App"
  type        = string
  default     = null
}

# Validate that exactly one of the two variables is set
variable "validate_one_set" {
  description = "Validation variable"
  default     = null

  validation {
    condition     = (var.app_service_name != null) != (var.function_app_name != null)
    error_message = "Exactly one of 'app_service_name' or 'function_app_name' must be set."
  }
}
```